tracer/test/Datadog.Trace.Tests/Sampling/SpanSamplingRuleTests.cs

```diff
@@ -69,99 +75,131 @@ public void BuildFromConfigurationString_ShouldHandle_MalformedData(string confi
+            VerifySingleRule(config, TestSpans.CartCheckoutSpan, true);
+            VerifySingleRule(config, TestSpans.AddToCartSpan, false);
+            VerifySingleRule(config, TestSpans.ShippingAuthSpan, false);
+            VerifySingleRule(config, TestSpans.ShippingRevertSpan, false);
+            VerifySingleRule(config, TestSpans.RequestShippingSpan, false);
         }
 
         [Fact]
         public void WildcardService_ShouldMatch_OnOperation()
         {
-            var config = "[{\"service\":\"*\", \"name\":\"authorize\", \"sample_rate\":0.5, \"max_per_second\":1000.5}]";
+            var config = """[{"service":"*", "name":"authorize", "sample_rate":0.5, "max_per_second":1000.5}]""";
 
-            VerifySingleRule(config, CartCheckoutSpan, false);
-            VerifySingleRule(config, AddToCartSpan, false);
-            VerifySingleRule(config, ShippingAuthSpan, true);
-            VerifySingleRule(config, ShippingRevertSpan, false);
-            VerifySingleRule(config, RequestShippingSpan, false);
+            VerifySingleRule(config, TestSpans.CartCheckoutSpan, false);
+            VerifySingleRule(config, TestSpans.AddToCartSpan, false);
+            VerifySingleRule(config, TestSpans.ShippingAuthSpan, true);
+            VerifySingleRule(config, TestSpans.ShippingRevertSpan, false);
+            VerifySingleRule(config, TestSpans.RequestShippingSpan, false);
         }
 
         [Fact]
         public void WildcardOperation_ShouldMatch_OnService()
         {
-            var config = "[{\"service\":\"shopping-cart-service\", \"name\":\"*\", \"sample_rate\":0.5, \"max_per_second\":1000.5}]";
+            var config = """[{"service":"shopping-cart-service", "name":"*", "sample_rate":0.5, "max_per_second":1000.5}]""";
+
+            VerifySingleRule(config, TestSpans.CartCheckoutSpan, true);
+            VerifySingleRule(config, TestSpans.AddToCartSpan, true);
+            VerifySingleRule(config, TestSpans.ShippingAuthSpan, false);
+            VerifySingleRule(config, TestSpans.ShippingRevertSpan, false);
+            VerifySingleRule(config, TestSpans.RequestShippingSpan, false);
+        }
+
+        [Fact]
+        public void WildcardOperation_ShouldMatch_OnResource()
+        {
+            var config = """[{"service":"*", "name":"*", "resource": "/api/users/*", "sample_rate":0.5, "max_per_second":1000.5}]""";
+
+            VerifySingleRule(config, TestSpans.CartCheckoutSpan, true);
+            VerifySingleRule(config, TestSpans.AddToCartSpan, false);
+            VerifySingleRule(config, TestSpans.ShippingAuthSpan, false);
+            VerifySingleRule(config, TestSpans.ShippingRevertSpan, true);
+            VerifySingleRule(config, TestSpans.RequestShippingSpan, true);
+        }
+
+        [Fact]
+        public void WildcardOperation_ShouldMatch_OnTags()
+        {
+            var config = """[{"service":"shopping-cart-service", "name":"*", "sample_rate":0.5, "max_per_second":1000.5}]""";
```

Comment:
It looks like this test case is identical to `WildcardOperation_ShouldMatch_OnService` instead of testing a "tags" rule

Reply:
> Using the `==`/`!=` operators to compare floating-point numbers is, generally, a bad idea. Floating point values are inherently inaccurate, and comparing them for exact equality is almost never the desired semantics. 

https://www.jetbrains.com/help/rider/CompareOfFloatsByEqualityOperator.html

Reply:
Oops, good catch!

Reply:
Fixed.